froide/document/auth.py

```diff
@@ -25,6 +27,8 @@ def has_perm(self, request):
         obj = ctx["object"]
         if obj.user == request.user:
             return True
+        if has_authenticated_access(obj, request, "read"):
+            return True
         return super().has_perm(request)
```

Comment:
This should return the result of `can_read_object` directly and not call `super`. The [parent `has_perm`](https://github.com/stefanw/django-crossdomainmedia/blob/master/crossdomainmedia/auth.py#L88-L101) does the standard Django permission check which is already run as part of `can_read_object` so on deny it would unnecessarily run and fail twice.